rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -95,17 +96,17 @@ trait RuleOrNodeReportingServiceImpl extends ReportingService {
       nodeIds <- confExpectedRepo.findCurrentNodeIds(ruleId)
       reports <- findRuleNodeStatusReports(nodeIds, Set(ruleId))
     } yield {
-      RuleStatusReport(ruleId, reports.toIterable)
+      val toKeep = reports.values.flatMap( _._2.filter( _.ruleId == ruleId))
+      RuleStatusReport(ruleId, toKeep)
     }
   }
 
    override def findNodeStatusReport(nodeId: NodeId) : Box[NodeStatusReport] = {
     for {
-      reports <- findRuleNodeStatusReports(Set(nodeId), Set())
-      report  =  reports.groupBy(_.nodeId).map { case(nodeId, reports) => NodeStatusReport(nodeId, reports) }.toSet
-      result  <- if(report.size == 1) Full(report.head) else Failure("Found bad number of reports for node " + nodeId)
+      reports    <- findRuleNodeStatusReports(Set(nodeId), Set())
+      (run, set) <- reports.get(nodeId)
```

Comment:
no, it's an option with get, but you're right, it should be better with an error message